js/pages/questionnaire.js

```diff
@@ -14,200 +13,194 @@ let modules;
 const questDiv = "questionnaireRoot";
 
 const importQuest = async () => {
+    const url = questConfig[location.host];
+    
+    try {    
+        const { transform } = await import(url);
 
-    let url = questConfig[location.host];
+        if (!transform) {
+            throw new Error('Transform error while importing quest');
+        }
 
-    await import(url).then(({ transform }) => {
         quest = transform;
-    });
+    } catch (e) {
+        throw new Error(`Error importing quest from ${url}: ${e.message}`)
+    }
 }
 
+/**
+ * Questionnaire directs the survey module handling throughout the survey loading process.
+ * Errors from children are caught here. The loading animation is shown and hidden here.
+ * @param {string} moduleId - The ID of the survey module the participant clicked to start.
+ */
 export const questionnaire = async (moduleId) => {
- 
-    showAnimation();
+    try {
+        showAnimation();
 
-    let responseData = await getMyData();
+        if (!moduleId) {
+            throw new Error('No module ID on start survey click.');
+        }
+
+        const responseData = await fetchDataWithRetry(() => getMyData());
+
+        if(!hasUserData(responseData)) {
+            throw new Error('No user data found.');
+        }
 
-    if(hasUserData(responseData)) {
         data = responseData.data;
 
         displayQuestionnaire(questDiv, moduleId !== 'ModuleSsn');
 
         if(moduleId === 'ModuleSsn') {
             socialSecurityTemplate(data);
-        }
-        else {
-
-            let responseModules = await getMySurveys([...new Set([fieldMapping.Module1.conceptId, fieldMapping.Module1_OLD.conceptId, fieldMapping.Biospecimen.conceptId, fieldMapping.ClinicalBiospecimen.conceptId, fieldMapping[moduleId].conceptId])]);
-            if(responseModules.code === 200) {
-                modules = responseModules.data;
+        } else {
+            const responseModules = await fetchDataWithRetry(() => getMySurveys([...new Set([
+                fieldMapping.Module1.conceptId,
+                fieldMapping.Module1_OLD.conceptId,
+                fieldMapping.Biospecimen.conceptId,
+                fieldMapping.ClinicalBiospecimen.conceptId,
+                fieldMapping[moduleId].conceptId
+            ])]));
+            
+            modules = responseModules.data;
 
-                await startModule(data, modules, moduleId, questDiv);
+            if (!modules) {
+                throw new Error('No modules found.');
```

Comment:
Did you test this with a participant that didn't have any module data? (aka, started their first module) Just want to make sure `modules` wouldn't evaluate as false if `responseModules.data` was an empty array.

Reply:
thanks for suggesting -- will test now

Reply:
Looks good:

New participant signup (UofC) ->
I was able to access and complete 'Background and Overall Health.'
SSN is also active on signup.
The other three modules are greyed out.

Upon completion of 'Background and Overall Health', the other three modules were activated:
'Where you Live and Work'
'Medications, Reproductive health, Exercise, and Sleep'
'Smoking, Alcohol, and Sun Exposure'

I was able to access and start all three.

Exiting and returning (continue button) ->
All remaining surveys loaded and started back up where I left off.